Kernel/Security/Jail.cpp

```diff
@@ -1,87 +0,0 @@
-/*
- * Copyright (c) 2022-2023, Liav A. <liavalb@hotmail.co.il>
- *
- * SPDX-License-Identifier: BSD-2-Clause
- */
-
-#include <AK/IntrusiveList.h>
-#include <AK/Singleton.h>
-#include <Kernel/API/Jail.h>
-#include <Kernel/Security/Jail.h>
-#include <Kernel/Tasks/Process.h>
-
-namespace Kernel {
-
-static Atomic<u64> s_jail_id;
-static Singleton<SpinlockProtected<Jail::List, LockRank::None>> s_all_instances {};
-
-static JailIndex generate_jail_id()
-{
-    return s_jail_id.fetch_add(1);
-}
-
-RefPtr<ProcessList> Jail::process_list()
-{
-    return m_process_list;
-}
-
-ErrorOr<NonnullRefPtr<Jail>> Jail::create(NonnullOwnPtr<KString> name, unsigned flags)
```

Comment:
I'm sure you had your reasons for this (and neither am I arguing in favor or against in particular), but doing a 180 on the whole feature like this is not exactly confidence-inspiring.

Reply:
I just realized that making everything tied to one object in the kernel is not the right solution, as it's not flexible enough. By introducing what is essentially our own implementation of Linux `namespaces`, we can achieve full isolation in a reasonable fashion and configuration of such container can happen incrementally instead of one syscall (`create_jail`) that you can't actually reason about its return value.

Reply:
Also, I'd like to put a note that jails are not entirely removed. It's just changing its purpose - it sets a new goal for it, so being jailed now is simply a kernel flag which restricts you from doing many things just like before (mainly dangerous syscalls but not just that). The other restrictions like PID isolation are moved elsewhere, because I'd like to build a flexible solution.